src/porepy/numerics/ad/operators.py

```diff
@@ -1750,88 +1802,111 @@ def __repr__(self) -> str:
             f"nodes ({self._nodes})\n"
         )
         if self.prev_iter:
-            s += "Evaluated at the previous iteration.\n"
+            s += f"Evaluated at the previous iteration {self.iterate_index}.\n"
         elif self.prev_time:
-            s += "Evaluated at the previous time step.\n"
+            s += f"Evaluated at the previous time step {self.time_step_index}.\n"
 
         return s
 
 
 class MixedDimensionalVariable(Variable):
     """Ad representation of a collection of variables that individually live on separate
-    subdomains or interfaces, but treated jointly in the mixed-dimensional sense.
+    subdomains or interfaces, but represent the same quantity and are treated jointly in
+    the mixed-dimensional sense.
+
+    Note:
+        As of now, the wrapped fixed-dimensional variables must fulfill the following
+        assumptions:
 
-    Conversion of the variables into numerical value should be done with respect to the
-    state of an array; see :meth:`Operator.evaluate`. Therefore, the MergedVariable does
-    not implement the method :meth:`Operator.parse`.
+        1. They have the same name
+        2. They are at the same time step and iterate.
+        3. They are defined in different grids (no overlaps).
 
     Parameters:
-        variables: List of variables to be merged. Should all have the same name.
+        variables: List of variables to be merged.
+
+    Raises:
+        AssertionError: If one of the above assumptions is violated.
 
     """
 
     def __init__(self, variables: list[Variable]) -> None:
-        ### PUBLIC
-
-        self.sub_vars = variables
-        """List of sub-variables passed at instantiation, each defined on a separate
-        domain.
 
-        """
+        time_indices = []
```

Comment:
The MD variable has now new assumptions: Atomic variables must not only have the same name, but be at the same instance of time and iterate.